packages/dumb/src/components/trends/cbg/cbg-slice-segment.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { FunctionComponent } from 'react'
+import { bindActionCreators, Dispatch } from 'redux'
+import { connect } from 'react-redux'
+import { CbgPositionData } from '../../../models/cbg-position-data.model'
+import { delayShowCbgTracesOnFocus, unfocusTrendsCbgSlice } from 'tidepool-viz'
+
+interface CbgSliceSegmentProps {
+  classes: string
+  datum: { id: string }
+  interpolated: { key: string, style: { [key: string]: string } }
+  positionData: CbgPositionData
+  segment: {
+    height: string
+    heightKeys: string[]
+    y: string
+  }
+  width: number
+  x: number
+  // Properties added via Redux
+  focusSlice: (userId: string, datum: { id: string }, positionData: CbgPositionData, focusedKeys: string[]) => void
+  unfocusSlice: (userId: string) => void
+  userId: string
+}
+
+const CBG_CIRCLE_PREFIX_ID = 'cbgCircle'
+
+const CbgSliceSegment: FunctionComponent<CbgSliceSegmentProps> = (props) => {
+  const { classes, datum, focusSlice, interpolated, positionData, segment, unfocusSlice, x, width, userId } = props
+
+  const key = interpolated.key
+  const style = interpolated.style
```

Comment:
I have trouble understanding what `interpolated` means or refers to.
I would suggest to passing the style as it owns props (not in an `interpolated` object), and instead of passing a key, you could pass an `id` (still not in `interpolated` object)?